File: SAMSUNG_SW_Expert_Academy/lv3/유효기간.py
# 10059
# 연습문제

'''
5
0721
2107
0507
0705
1313
'''

testcase = int(input())
answer = []
for tc in range(testcase):
    number = input().rstrip()
    number = [int(number[:2]), int(number[2:])]
    result = 'NA'
    if 1 <= number[0] <= 12:
        result = 'MMYY'
    if 1 <= number[1] <= 12:
        if result == 'NA':
            result = 'YYMM'
        else:
            result = 'AMBIGUOUS'
    
    answer.append(f'#{tc+1} {result}')

for ans in answer:
    print(ans)

File: SAMSUNG_SW_Expert_Academy/lv3/유효기간.java
//10059
//연습문제

import java.io.*;

/*
5
0721
2107
0507
0705
1313
*/

public class 유효기간 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {
        int testcase = Integer.parseInt(br.readLine());
        for (int tc = 1; tc <= testcase; tc++) {
            String number = br.readLine().replace(" ", "");
            Integer[] _number = {
                Integer.parseInt(number.substring(0,2)),
                Integer.parseInt(number.substring(2,4))
            };
            String result = "NA";
            if (1 <= _number[0] && _number[0] <= 12){
                result = "MMYY";
            }
            if (1 <= _number[1] && _number[1] <= 12){
                if (result == "NA"){
                    result = "YYMM";
                } else {
                    result = "AMBIGUOUS";
                }
            }
            sb.append("#"+tc+" "+result+"\n");
        }
        System.out.print(sb);
    }
}